src/modules/vendor/components/questionnaires/ProfileAnswers.vue

```diff
@@ -1,29 +1,64 @@
 <template>
   <div>
-    <q-card v-for="(card, cardIndex) of questionnaireAnswers.followUp" :key="cardIndex" flat class="q-mb-sm">
+    <ni-select :options="trainerList" :value="selectedTrainer" @input="updateSelectedTrainer" class="select" />
+    <q-card v-for="(card, cardIndex) of questionnaireAnswersFormattedAndFiltered.followUp" :key="cardIndex" flat
+      class="q-mb-sm">
       <component :is="getChartComponent(card.template)" :card="card" />
     </q-card>
   </div>
 </template>
 
 <script>
+import uniqBy from 'lodash/uniqBy';
+import get from 'lodash/get';
 import Questionnaires from '@api/Questionnaires';
+import Users from '@api/Users';
+import Select from '@components/form/Select';
+import { formatAndSortIdentityOptions } from '@helpers/utils';
 import { NotifyNegative } from '@components/popup/notify';
 import { questionnaireAnswersMixin } from '@mixins/questionnaireAnswersMixin';
+import { TRAINER, TRAINING_ORGANISATION_MANAGER, VENDOR_ADMIN } from '@data/constants';
 
 export default {
   name: 'ProfileAnswers',
+  components: {
+    'ni-select': Select,
+  },
   mixins: [questionnaireAnswersMixin],
   props: {
     profileId: { type: String, required: true },
   },
   data () {
     return {
-      questionnaireAnswers: [],
+      questionnaireAnswers: {},
+      selectedTrainer: '',
+      trainerList: [],
     };
   },
   async created () {
-    await this.getQuestionnaireAnswers();
+    await Promise.all([this.getQuestionnaireAnswers(), this.getTrainerList()]);
+  },
+  computed: {
+    questionnaireAnswersFormattedAndFiltered () {
+      let followUp = [];
+      if (!get(this.questionnaireAnswers, 'followUp')) return {};
```

Comment:
dans la logique ici tu pourrais retourner `...this.questionnaireAnswers,`

Reply:
est-ce-que c'est pas un contresens sachant que ce if sert à éviter de renvoyer une erreur quand this.questionnaireAnswers est pas encore défini ?